error when using ocaml-gmp with mirage 4.2.0 on FreeBSD and solo5-hvt

When I try to compile from mirage-skeleton the application/static_website_tls, I get the following error:
```
static_website_tls# dune bu
Done: 99% (6110/6125, 15 left) (jobs: 1)ELF interpreter /nonexistent/solo5/ not found, error 2
ELF interpreter /nonexistent/solo5/ not found, error 2
File "duniverse/ocaml-gmp/src/dune", line 14, characters 0-319:
14 | (rule
15 |  (targets gmp.h libgmp.a dllgmp.so)
16 |  (deps gmp-6.2.1.tar.xz build.sh)
17 |  (action
18 |   (with-stdout-to
19 |    build.log
20 |    (run sh ./build.sh "%{ocaml-config:c_compiler}" "%{ocaml-config:host}"
21 |      "%{ocaml-config:ocamlc_cflags}"
22 |      %{ocaml-config:supports_shared_libraries}
23 |      "%{ocaml-config:native_c_libraries}"))))
configure: error: Cannot find a build system compiler
```

A verbose run of dune shows:
```
Command [18] exited with code 1:
$ (cd _build/solo5/duniverse/ocaml-gmp/src && /bin/sh ./build.sh x86_64-solo5-none-static-cc x86_64-unknown-none '-I/usr/home/hannes/.opam/mirage4/solo5-sysroot/include/nolibc/ -include _solo5/overrides.h -O2 -fno-strict-aliasing -fwrapv' false '-L/usr/home/hannes/.opam/mirage4/solo5-sysroot/lib/nolibc/ -lnolibc -lopenlibm') > _build/solo5/duniverse/ocaml-gmp/src/build.log
configure: error: Cannot find a build system compiler
```

And a manual run thereof results in:
```
static-website-tls# cd _build/solo5/duniverse/ocaml-gmp/src
src# /bin/sh ./build.sh x86_64-solo5-none-static-cc x86_64-unknown-none '-I/usr/home/hannes/.opam/mirage4/solo5-sysroot/include/nolibc/ -include _solo5/overrides.h -O2 -fno-strict-aliasing -fwrapv' false '-L/usr/home/hannes/.opam/mirage4/solo5-sysroot/lib/nolibc/ -lnolibc -lopenlibm'
checking build system type... x86_64-unknown-freebsd13.0
checking host system type... x86_64-unknown-none
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for x86_64-unknown-none-strip... no
checking for strip... strip
checking for a thread-safe mkdir -p... ./install-sh -c -d
checking for gawk... no
checking for mawk... no
checking for nawk... nawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking ABI=64
checking whether x86_64-solo5-none-static-cc is gcc... yes
checking compiler x86_64-solo5-none-static-cc -O2 -pedantic -fomit-frame-pointer -m64 -I/usr/home/hannes/.opam/mirage4/solo5-sysroot/include/nolibc/ -include _solo5/overrides.h -O2 -fno-strict-aliasing -fwrapv... yes
checking compiler x86_64-solo5-none-static-cc -O2 -pedantic -fomit-frame-pointer -m64 -I/usr/home/hannes/.opam/mirage4/solo5-sysroot/include/nolibc/ -include _solo5/overrides.h -O2 -fno-strict-aliasing -fwrapv -mtune=k8... yes
checking compiler x86_64-solo5-none-static-cc -O2 -pedantic -fomit-frame-pointer -m64 -mtune=k8 -I/usr/home/hannes/.opam/mirage4/solo5-sysroot/include/nolibc/ -include _solo5/overrides.h -O2 -fno-strict-aliasing -fwrapv -march=k8... yes
checking for x86_64-unknown-none-gcc... x86_64-solo5-none-static-cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... yes
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether x86_64-solo5-none-static-cc accepts -g... yes
checking for x86_64-solo5-none-static-cc option to accept ISO C89... none needed
checking whether x86_64-solo5-none-static-cc understands -c and -o together... yes
checking for x86_64-solo5-none-static-cc option to accept ISO C99... unsupported
checking how to run the C preprocessor... x86_64-solo5-none-static-cc -E
checking build system compiler x86_64-solo5-none-static-cc... ELF interpreter /nonexistent/solo5/ not found, error 2
no
checking build system compiler x86_64-solo5-none-static-cc -O2 -pedantic -fomit-frame-pointer -m64 -mtune=k8 -march=k8 -I/usr/home/hannes/.opam/mirage4/solo5-sysroot/include/nolibc/ -include _solo5/overrides.h -O2 -fno-strict-aliasing -fwrapv... ELF interpreter /nonexistent/solo5/ not found, error 2
no
checking build system compiler cc... no
checking build system compiler gcc... no
checking build system compiler c89... no
checking build system compiler c99... no
configure: error: Cannot find a build system compiler
```

Any idea what went wrong and how to fix this @TheLortex?

--- a/src/build.sh.in
+++ b/src/build.sh.in
@@ -4,8 +4,8 @@ set -e
 
 tar -xf gmp-6.2.1.tar.xz
 cd gmp-6.2.1
-sed -i.bak 's/\(gmp_compile=.*conftest.c\)/\1 $LIBS/' configure
-sed -i.bak 's/\(gmp_compile="\($HOST_CC\|$i\|$CC_FOR_BUILD\).*conftest.c\) $LIBS/\1/' configure
+sed -i -e 's/\(gmp_compile="$cc .*conftest.c\)/\1 $LIBS/g' configure
+sed -i -e 's/\(gmp_compile="$CC .*conftest.c\)/\1 $LIBS/g' configure
 
 if [ "$4" = "false" ]; then
     SHARED_LIBRARY_ARG="--disable-shared"

